src/forLoopHelpers.ts

```diff
@@ -108,7 +113,14 @@ export const getForLoopSyntaxResult = (forLoopSyntax: ForLoopSyntax): string =>
       forLoopSyntax.x,
       i.toString()
     )
-    forLoopResult = replaceAll(forLoopResult, `[${forLoopSyntax.x}]`, i.toString())
+
+    if (typeof i === 'object') {
+      Object.keys(i).forEach(key => {
+        forLoopResult = replaceAll(forLoopResult, `[${forLoopSyntax.x}.${key}]`, (i as IObject)[key].toString())
```

Comment:
Here you are not really sure that `i` is of type `IObject` and will failed if the user use an array or another object. I like the idea of only allowing a single level of deepness in the object to keep it simple. However it would be nice to test the `i` variable with a new function  `isStatementObjectValid` and throw an Error if not like it’s done here https://github.com/johannchopin/restapify/blob/main/src/Restapify.ts#L220.

This function will then only test that for each key the value is of type `string | number | boolean` 👍 

What's your opinion on this idea? Do you have time for that or should I do it?

Reply:
Okay, I give up on this one! I admittedly am new to Typescript and was bluffing my way through 😅, but my Google-fu wasn't good enough to help me implement this particular type check. Could you please suggest an implementation for this? 🙏

Reply:
Héhé no worry you are already on the right path 👍  So the idea is to have this following util function...

```ts
const isStatementObjectValid = (obj: Record<string, any>): boolean  => {
  return Object.keys(obj).every(key => {
    return typeof obj[key] === 'string' ||
      typeof obj[key] === 'number' ||
      typeof obj[key] === 'boolean'
  })
}
```

...and then before calling the `replaceAll` you make sure that `i` is valid like:

```ts
    const isStatementValid = isStatementObjectValid(i)
    if (!isStatementValid) /* Throw an error here like INV:SYNTAX */
    else {
        Object.keys(i).forEach(key => {
            forLoopResult = replaceAll(forLoopResult, `[${forLoopSyntax.x}.${key}]`, (i as IObject)[key].toString())
    }
```

Like that you will still need the type cast `(i as IObject)` but your are programmatically sure that the object is valid. Concerning the error throwing I'm not sure yet so I will try to implement something in your fork directly if it's fine for you.

Reply:
Ah thanks, that makes sense! I've committed the `isStatementObjectValid` function. Please feel free to implement the error throwing directly in the fork.

Reply:
So I added the check that will throw a runtime error 👍  i fix the coverage issue and then we can merge this one 🚀